### DISCUSSION TITLE
Bad inference for short sentences

### DISCUSSION THREAD

**Discussion by cdlIris** [coqui-ai/TTS#963] [General Q&A]
Hi,

I have been encountered this problem, while testing with the pre-trained model, and the model that I trained with my own dataset. When the sentences contain 2 or more words, it will work correctly most of the time. However, when the sentence is 1 word, like "Hello!", "Hi", it will generate unintelligible result.

Here are some plots of sentence "Hello!":
Those are mel_outputs, mel_outputs_postnet, alignments

**Some Background:**
- has been trained 101k
- Train loss 0.146386, Grad Norm 0.442730, Validation loss: 0.105691

![image](https://user-images.githubusercontent.com/38269254/143324628-660ebb1b-64cc-4c52-bb40-2de7caa1aa52.png)

**Comment by thorstenMueller** [comment]
Does your own dataset include one word phrases? Does a ! or a . make a difference as last character?

**Reply by erogol** [reply]
Which pre-trained model did not work?

**Reply by cdlIris** [reply]
It was the English tactoron2-ddc model, that trained with ljspeech dataset.

**Reply by cdlIris** [reply]
> Does your own dataset include one word phrases? Does a ! or a . make a difference as last character?

Yes, I have checked my dataset, I have a few data lines that are one word phrases (less than 20 data lines). If I want to make it better, I probably need to increase data with this pattern. However, how much new lines to add is reasonable? Thank you!

**Reply by erogol** [reply]
Don't have a certain number but if one-word samples are important for you to add as many samples as possible. 

From the spectrogram, it looks like the model reads the word but does not stop. Is this the case ?
